pkg/shardddl/optimism/lock.go

```diff
@@ -121,6 +123,8 @@ func (l *Lock) TrySync(callerSource, callerSchema, callerTable string,
 	// if any real conflicts after joined exist, they will be detected by the following steps.
 	var cmp int
 	if cmp, err = newTable.Compare(oldJoined); err == nil && cmp == 0 {
+		_, remain := l.syncStatus()
+		l.synced = remain == 0
```

Comment:
How about doing this in `defer`?

Reply:
addressed in https://github.com/pingcap/dm/pull/814/commits/95408c3290ff554c652fd44f737ce9a82ffbc2df